test/Dialect/Seq/canonicalization.mlir

```diff
@@ -87,3 +87,36 @@ hw.module @UninitializedArrayElement(%a: i1, %clock: i1) -> (b: !hw.array<2xi1>)
   %1 = hw.array_create %0, %a : i1
   hw.output %r : !hw.array<2xi1>
 }
+
+// CHECK-LABEL: hw.module @ClockGate
+hw.module @ClockGate(%clock: i1, %enable: i1, %testEnable: i1) {
+  // CHECK-NEXT: hw.constant false
+  %false = hw.constant false
+  %true = hw.constant true
+
+  // CHECK-NEXT: %zeroClock = hw.wire %false sym @zeroClock
+  %0 = seq.clock_gate %false, %enable
+  %zeroClock = hw.wire %0 sym @zeroClock : i1
+
+  // CHECK-NEXT: %alwaysOff1 = hw.wire %false sym @alwaysOff1
+  // CHECK-NEXT: %alwaysOff2 = hw.wire %false sym @alwaysOff2
+  %1 = seq.clock_gate %clock, %false
+  %2 = seq.clock_gate %clock, %false, %false
```

Comment:
Using true for the enable would provide more coverage.

Reply:
There are separate always-on tests for the constant true case. The problem here is if I pass one of the enables as constant true, it will just make the clock gate fold away (the enables are ORed).